app/controllers/subscriptions_management_controller.rb

```diff
@@ -118,4 +118,12 @@ def is_single_page_subscription?(subscription)
     subscription.dig("subscriber_list", "content_id").present?
   end
   helper_method :is_single_page_subscription?
+
+  def get_updated_at_from_subscription(subscription)
+    base_path = subscription["subscriber_list"]["url"]
+    content_item = GdsApi.content_store.content_item(base_path)
+
+    content_item["public_updated_at"]
+  end
+  helper_method :get_updated_at_from_subscription
```

Comment:
Interesting, so we're putting a lot of helper methods in the controller.
I think we can do that, it's useful when you want to use these methods as part of controller actions. So perhaps before you send a `@user` variable over to the view we use a method in the controller..

Then for whatever reason we might also want it to be separately available in the view. So we put it in here with that `helper_method` tag.

I _think_ if there's not a need for it to go in the controller, the correct place is in `app/helpers`.

These are cool, they're just really simple ruby modules ([our accounts one](https://github.com/alphagov/email-alert-frontend/blob/main/app/helpers/account_helper.rb) is a good example), that just define methods.

Rails magic then sweeps them up and automatically includes them into the views. It doesn't matter what we name these as [include_all_helpers defaults to true now](https://api.rubyonrails.org/classes/ActionController/Helpers.html)

So we could make something like:
```
# app/helpers/subscriptions_herlper.rb
module SubscriptionsHelper
  def get_updated_at_from_subscription(subscription)
    ...
  end
end
```

And it should automagically work in your view
And then yo

Reply:
I've moved this out into a helper. I'm not quite sure what the test should look like though

Reply:
Ok, I've put in a simple helper test as well now. I've left the one in the controller as I think that's testing how we use the helper rather than the helper itself.